internal/controller/etcd/config.go

```diff
@@ -0,0 +1,115 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package etcd
+
+import (
+	"errors"
+	"time"
+
+	"github.com/gardener/etcd-druid/internal/controller/utils"
+	"github.com/gardener/etcd-druid/internal/features"
+	flag "github.com/spf13/pflag"
+	"k8s.io/component-base/featuregate"
+)
+
+// Flag names
+const (
+	workersFlagName                            = "etcd-workers"
+	ignoreOperationAnnotationFlagName          = "ignore-operation-annotation"
+	enableEtcdSpecAutoReconcileFlagName        = "enable-etcd-spec-auto-reconcile"
+	disableEtcdServiceAccountAutomountFlagName = "disable-etcd-serviceaccount-automount"
+	etcdStatusSyncPeriodFlagName               = "etcd-status-sync-period"
+	etcdMemberNotReadyThresholdFlagName        = "etcd-member-notready-threshold"
+	etcdMemberUnknownThresholdFlagName         = "etcd-member-unknown-threshold"
+)
+
+const (
+	defaultWorkers                            = 3
+	defaultIgnoreOperationAnnotation          = false
+	defaultDisableEtcdServiceAccountAutomount = false
+	defaultEnableEtcdSpecAutoReconcile        = false
+	defaultEtcdStatusSyncPeriod               = 15 * time.Second
+	defaultEtcdMemberNotReadyThreshold        = 5 * time.Minute
+	defaultEtcdMemberUnknownThreshold         = 1 * time.Minute
+)
+
+// featureList holds the feature gate names that are relevant for the Etcd Controller.
+var featureList = []featuregate.Feature{
+	features.UseEtcdWrapper,
+}
+
+// Config defines the configuration for the Etcd Controller.
+type Config struct {
+	// Workers is the number of workers concurrently processing reconciliation requests.
+	Workers int
+	// IgnoreOperationAnnotation specifies whether to ignore or honour the operation annotation on resources to be reconciled.
+	// Deprecated: Use EnableEtcdSpecAutoReconcile instead.
```

Comment:
Could we create an issue to track and plan the removal of the deprecated field `IgnoreOperationAnnotation` in the near future?

Reply:
Created #801 to track removal of this and `Status.LastError`.